### DISCUSSION TITLE
How to set a custom cursor when user's mouse is not over the app's window?

### DISCUSSION THREAD

**Discussion by MartyIX** [microsoft/WindowsAppSDK#3320] [Q&A]
WinUI 3 allows to drag and drop between an object between two windows of a single application. I have an application (it's a [MAUI](https://github.com/dotnet/maui/) application but my issue is more related to WinUI 3) where I modify cursor caption (https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.draguioverride.caption?view=winrt-22621) like this:

```cs
// Override the default hints displayed while dragging an element.
Microsoft.Maui.Handlers.ElementHandler.ElementMapper.AppendToMapping("DragDrop.DragOverUi", (handler, view) =>
{
    if (view is View mauiView)
    {
        if (handler.PlatformView is Microsoft.UI.Xaml.UIElement uiElement)
        {
            DropGestureRecognizer? dropGestureRecognizer = mauiView.GestureRecognizers.OfType<DropGestureRecognizer>().FirstOrDefault();
            if (dropGestureRecognizer is not null)
            {
                string? caption = Handlers.GetDragUiCaption(dropGestureRecognizer);
                if (caption is not null)
                {
                    uiElement.DragEnter += (object sender, Microsoft.UI.Xaml.DragEventArgs e) =>
                    {
                        if (string.IsNullOrEmpty(caption))
                        {
                            e.DragUIOverride.IsCaptionVisible = false;
                            e.DragUIOverride.IsGlyphVisible = false;
                        }
                        else
                        {
                            e.DragUIOverride.Caption = caption; // This is the important part.
                            e.DragUIOverride.IsCaptionVisible = true;
                        }
                    };
                }
            }
        }
    }
});
```

and it works great but my cursor changes back to the default once user's moves her mouse out of application's window. In other words, `DragUIOverride.Caption` applies only when mouse pointer is over my application windows.

Is there any way to fix it?

Thank you

PS: I have found https://github.com/microsoft/WindowsAppSDK/discussions/2722 but it looks like a different discussion to me.

**Comment by DarranRowe** [comment]
Functions like [SetCursor](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setcursor) indicate that an application shouldn't set the cursor unless the cursor is within one of the application's windows or it is capturing mouse input.
Since WinUI 3 is still based on the desktop environment, I would at least suggest you try doing something similar. [UIElement.CapturePointer](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.uielement.capturepointer?view=windows-app-sdk-1.2#microsoft-ui-xaml-uielement-capturepointer(microsoft-ui-xaml-input-pointer)) captures the pointer.

**Reply by MartyIX** [reply]
Thank you for taking time to reply.

I have not found a way to access `pointer` in `DragEnter` event to set pointer capture. Is there any?

**Reply by DarranRowe** [reply]
Unfortunately, with how the WinRT Xaml does things, the only way to obtain a pointer is through the PointerRoutedEventArgs. Because WinUI copied this, it also means that the only way to get the pointer in WinUI 3 is through the PointerRoutedEventArgs.
So your code either needs to set flags to keep track of the state, or you need to cache the pointer from one of the pointer event handlers. But also remember, at most, you only need to have the pointer captured while it is outside of your application's windows. This means that you don't have to capture it right away.
